python/pyspark/sql/tests/test_functions.py

```diff
@@ -1003,6 +1003,30 @@ def test_np_scalar_input(self):
             res = df.select(array_position(df.data, dtype(1)).alias("c")).collect()
             self.assertEqual([Row(c=1), Row(c=0)], res)
 
+    @unittest.skipIf(not have_numpy, "NumPy not installed")
+    def test_ndarray_input(self):
+        import numpy as np
+
+        int_arrs = [np.array([1, 2]).astype(t) for t in ["int8", "int16", "int32", "int64"]]
+        for arr in int_arrs:
+            self.assertEqual(
+                [Row(b=[1, 2])], self.spark.range(1).select(lit(arr).alias("b")).collect()
+            )
+
+        float_arrs = [np.array([0.1, 0.2]).astype(t) for t in ["float32", "float64"]]
+
+        self.assertEqual(
+            [("b", "array<double>")],
+            self.spark.range(1).select(lit(float_arrs[0]).alias("b")).dtypes,
+        )
+        self.assertEqual(
```

Comment:
We cannot compare the Row equality by calling `collect` with the error showed below:
due to 
```
AssertionError: Row(b=[0.10000000149011612, 0.20000000298023224]) != [Row(b=[0.10000000149011612, 0.20000000298023224])]
```

We compare `dtypes` and the actual data in the Row separately.

Reply:
My understanding is array functions like `concat` / `array_intersect` would not accept numpy array input.
Do they accept Java arrays in Scala side?

Reply:
yes, they should be in a `Literal`. Nvm

Reply:
Thank you!